core/field_variable.js

```diff
@@ -163,9 +163,11 @@ Blockly.FieldVariable.dropdownCreate = function() {
   }
   variableModelList.sort(Blockly.VariableModel.compareByName);
   var options = [];
-  for (var i = 0; i < variableModelList.length; i++) {
+  for (var i = 0, variable; variable = variableModelList[i]; i++) {
+    if (!variable.disabled || this.getValue() == variable.name) {
```

Comment:
You check ` if(!variable.disabled)` quite a few times in this PR.  

It would be easier to parse `if (variable.isEnabled())`.  Also (per offline discussion) switch to storing `enabled` instead of `disabled` to avoid double negatives.